snuba/web/query.py

```diff
@@ -285,44 +280,14 @@ def parse_and_run_query(
     return result
 
 
-@split_query
-def _run_query(
+def _run_clickhouse_query(
```

Comment:
Most of this function is not ClickHouse specific… we earlier were trying to make as much of this storage agnostic as possible but this implies we're going the other way?

Reply:
I tried splitting this file in different ways and in the end  I am not sure I agree with the point that the responsibilities are not well defined at least for most methods. We have:
- run the query with caching deduping and rate limiting
- run the query with stats recording
- run the query with APM instrumentaion
- process query and run. 

Apart from splitting caching, deduping, and rate limiting from actually running the query, I  would not make major changes to the split of responsibilities. I really think we have a vocabulary problem there.

This does not mean there is no opportunity to improve this class, but I don't think the main issue is the split of responsibilities.

Reply:
Let me try one last change, hold on on this comment.

Reply:
I am about to send a separate PR based on this. I would not be comfortable in deploying this change and the one I am making together, too risky.